source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -37,6 +37,8 @@ void OnDemandRouteUpdate::setDecoderFilterCallbacks(Http::StreamDecoderFilterCal
   callbacks_ = &callbacks;
 }
 
+void OnDemandRouteUpdate::onDestroy() { route_config_updated_callback_.reset(); }
```

Comment:
Thanks for adding a test for this! 

So you're resetting the callback, but what tells the caller of the callback to not call it?
you've created this lambda capturing *this* of the filter, and if the filter is destroyed, and the resolution completes, won't the callback still be called?

Reply:
The callback pointer is eventually stored in a weak_ptr, which is verified before the invocation: https://github.com/envoyproxy/envoy/blob/master/source/common/router/rds_impl.cc#L288. A longer description of what's going on in relation to the callback is here: https://github.com/envoyproxy/envoy/issues/9784#issuecomment-635034755.